src/interval.jl

```diff
@@ -161,7 +181,8 @@ end
 
 ##### ARITHMETIC #####
 
-Base.:+(a::T, b) where {T <: Interval} = T(first(a) + b, last(a) + b, inclusivity(a))
+Base.:+(a::Interval, b) = Interval(first(a) + b, last(a) + b, inclusivity(a))
```

Comment:
The previous version always kept the `Interval` unit type the same. This could cause the Interval type to change.

Reply:
This becomes the same issue as most of the other issues that cropped up by adding Infinity. If `a` is `-∞ .. ∞` and `b` is of type `Int` then `∞ - b` where `b` is `1` becomes an `InfExtended{Int64}` type which will error out since T would have been `Interval{Infinite}(...)`. So we end up trying to run `convert(Infinite, InfExtended{Int64})` which is undefined. That being said, maybe that shouldn't be undefined. Converting an `InfExtended{Int64})` to `Infinite` could potentially work if we check the value to make sure it's not an actual extended value

Reply:
That conversion makes sense if the `InfExtended{Int64}` is infinite

Reply:
Fixed by adding the conversion: https://github.com/cjdoris/Infinity.jl/pull/7